haystack/reader/farm.py

```diff
@@ -233,19 +234,7 @@ def train(
             use_amp=use_amp,
         )
         # 4. Feed everything to the Trainer, which keeps care of growing our model and evaluates it from time to time
-        trainer = Trainer(
-            model=model,
-            optimizer=optimizer,
-            data_silo=data_silo,
-            epochs=n_epochs,
-            n_gpu=n_gpu,
-            lr_schedule=lr_schedule,
-            evaluate_every=evaluate_every,
-            device=device,
-            use_amp=use_amp,
-            disable_tqdm=not self.progress_bar
-        )
-
+        trainer = create_or_load_checkpoint()
```

Comment:
This is the right spot to put create_or_load_checkpoint(), but you will need to pass all of the args that were used in Trainer() above + `checkpoint_root_dir` . Please also test the approach once yourself of training, stopping, resuming (can be in a colab notebook or similar).